cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -2025,6 +2069,37 @@ pub(crate) mod tests {
         assert_eq!(seed.packages, expected,);
     }
 
+    #[tokio::test]
+    async fn test_locking_unknown_message() {
+        let manifest = &*TEST_TYPED_MANIFEST;
+
+        let mut client = catalog::MockClient::new(None::<String>).unwrap();
+        let response = TEST_RESOLUTION_RESPONSE_UNKNOWN_MSG.clone();
+        let response_msg: ResolutionMessage =
+            response.first().unwrap().msgs.first().unwrap().clone();
+        client.push_resolve_response(response);
+
+        let locked_manifest = LockedManifestCatalog::lock_manifest(manifest, None, &client).await;
+        match locked_manifest {
+            Ok(_) => panic!(),
+            Err(LockedManifestError::ResolutionFailed(res_failures)) => {
+                assert_eq!(res_failures.0.len(), 1);
+                match &res_failures.0[0] {
+                    ResolutionFailure::UnknownServiceMessage {
+                        level,
+                        msg,
+                        context: _,
+                    } => {
+                        assert_eq!(msg, &response_msg.msg());
+                        assert_eq!(level, "error");
+                    },
+                    _ => panic!(),
+                }
+            },
+            _ => panic!(),
+        }
```

Comment:
Looks good! This is a little bit more concise, but some of it is just personal preference (e.g. if you want to be explicit about all 3 cases the match is just fine)
```suggestion
        if let Err(LockedManifestError::ResolutionFailed(res_failures)) = locked_manifest {
            if let [ResolutionFailure::UnknownServiceMessage {
                level,
                msg,
                context: _,
            }] = res_failures.0.as_slice()
            {
                assert_eq!(msg, &response_msg.msg());
                assert_eq!(level, "error");
            } else {
                panic!();
            }
        } else {
            panic!();
        }
```

Reply:
Great learning, thanks!  Re-wrote it myself and pushed after reading yours for the experience.